airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -319,11 +320,34 @@ def post_dag_run(*, dag_id: str, session: Session = NEW_SESSION) -> APIResponse:
     if not dagrun_instance:
         try:
             dag = get_airflow_app().dag_bag.get_dag(dag_id)
+
+            data_interval_start_exists = post_body.get("data_interval_start") is not None
+            data_interval_end_exists = post_body.get("data_interval_end") is not None
+
+            if (
+                data_interval_start_exists
+                and not data_interval_end_exists
+                or data_interval_end_exists
+                and not data_interval_start_exists
+            ):
+                raise BadRequest(
+                    detail="Both 'data_interval_start' and 'data_interval_end' must be specified, you cannot specify only one",
+                )
```

Comment:
Is it a way to validate in the schema instead these two should always both exist or be absent? I feel there should be a way.

Reply:
hmm, it makes sense, but I am not familiar with the codebase much, therefore I am looking forward to any pointers you might have. I couldn't think of any such examples off the top of my head to check the code.

Reply:
I am not 100% sure which schema version is used, but maybe taking a look here helps: https://json-schema.org/understanding-json-schema/reference/conditionals#dependentRequired

Reply:
are these json schema objects? it seems like a different kind of SQLAlchemy schema, does it support json schema spec?

Reply:
The OpenAPI schema is a derivative of JSON schema. There’s a compatibility table somewhere on the OpenAPI website. If it supports the syntax, it goes into `openapi/v1.yaml`, which is used in the API client (e.g. web front end).

But to validate at the back end, you need to change the Marshmallow declaration (in `schemas/dag_run_schema.py`). This should be a bit easier (in the sense of figuring out compatibility and whatnot). A [schema-level validation function](https://marshmallow.readthedocs.io/en/latest/extending.html#schema-level-validation) should work for this.

Reply:
@uranusjr please give it a look, I've added a schema-level validation.